src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
```

Comment:
@yannick-polius : I think the ```void*``` passing will not work if we are using this across processes. It's fine for now since it's only inside the ProbeReader,. If I understand this right then the void* is pointer to  a package right? This  would  cause memory corruption across process memory barriers since the void* is just a pointer. The pointer would be copied but would point to the wrong thing once it's received on the other process side. 

I could be wrong here and if so let's go through this in person.